module/os/freebsd/zfs/zfs_file_os.c

```diff
@@ -285,6 +285,35 @@ zfs_file_fsync(zfs_file_t *fp, int flags)
 	return (zfs_vop_fsync(fp->f_vnode));
 }
 
+/*
+ * deallocate - zero and/or deallocate file storage
+ *
+ * fp - file pointer
+ * offset - offset to start zeroing or deallocating
+ * len - length to zero or deallocate
+ */
+int
+zfs_file_deallocate(zfs_file_t *fp, loff_t offset, loff_t len)
+{
+	int rc;
+#if __FreeBSD_version >= 1400028
```

Comment:
It seems should be "1400029", according to https://github.com/freebsd/freebsd-src/commit/245ec7651e4221043d1032fb3f82f335dc65fc7f.

Reply:
Ahh, thanks. I can't remember where I got 1400028, but what you say makes sense. Fixed.